packages/schema-stitching/src/stitch/typeFromAST.ts

```diff
@@ -119,13 +121,9 @@ function makeScalarType(node: ScalarTypeDefinitionNode): GraphQLScalarType {
   return new GraphQLScalarType({
     name: node.name.value,
     description: getDescription(node, backcompatOptions),
-    serialize: () => null,
-    // Note: validation calls the parse functions to determine if a
-    // literal value is correct. Returning null would cause use of custom
-    // scalars to always fail validation. Returning false causes them to
-    // always pass validation.
-    parseValue: () => false,
-    parseLiteral: () => false,
+    serialize: value => value,
+    parseValue: value => value,
+    parseLiteral: value => value,
```

Comment:
Why do we need this change because `parseLiteral` will return ast node by default instead of graphqljs's default function?

Reply:
These 3 constructor values should just be dropped.

The flow types are optional (https://github.com/graphql/graphql-js/blob/master/src/type/definition.js#L653-L658) but the TypeScript types are not all yet (https://github.com/graphql/graphql-js/blob/master/src/type/definition.d.ts#L330-L335).

So will remove parseLiteral and parse, and keep default serialize, with note to change once graphql v16 with TypeScript changes lands.

Goal is to make typeFromAst build types in the exact same way as graphql-js -- this needs to be as similar as possible, because any changes upstream will have to manually added.

Reply:
I see but parseLiteral takes an astNode as an argument not a value so graphqljs's default parseLiteral implementation parses that;
https://github.com/graphql/graphql-js/blob/master/src/type/definition.js#L584

That means our version should pass that astNode to `valueFromASTUntyped` then parse it using `parseValue` as graphqljs currently does.